en/advanced-topics/jsb-auto-binding.md

```diff
@@ -0,0 +1,374 @@
+
+## Environment Configuration and Auto-Binding Demonstration
+
+### Environment Configuration
+
+Auto-binding, to put it simply, is a matter of executing a python script to automatically generate the corresponding `.cpp` and `.h` files. First,make sure that the computer has a python runtime environment. Example of how to install it on a Mac:
+
+1. To install python 3.0, download the installation package from the python website:
+
+    <https://www.python.org/downloads/release/python-398/>
+
+2. Install some of the python dependencies via pip3:
+
+    ```shell
+    sudo pip3 install pyyaml==5.4.1
+    sudo pip3 install Cheetah3
+    ```
+
+3. Install the NDK, which is definitely essential when it comes to C++. It is recommended to install the [Android NDK r16b](https://developer.android.com/ndk/downloads/older_releases?hl=zh-cn) version, set the `PYTH_profile` in `~/.bash_ profile`, then set `PYTHON_ROOT` and `NDK_ROOT` in `~/.bash_profile`, because these two environment variables will be used directly in the python file that will be executed later.
+
+    ```shell
+    export NDK_ROOT=/Users/kevin/android-ndk-r16b
+    export PYTHON_BIN=python3
+    ```
+
+In Windows, directly install the modules by referring to the above, and remember to configure the environment variables at the end.
+
+### Auto-Binding Demonstration
+
+Here's a demonstration of how the files under the cocos engine, i.e. the **cocos/bindings/auto** directory (shown below) are automatically generate:
+
+![](jsb/auto-file.png)
+
+In fact, the beginning of these file names that there are some specific rules for naming these files, so how are these files generated? First, open a terminal, `cd` to the **tools/tojs** directory, and then run `./genbindings.py`.
+
+![](jsb/generate-file.png)
+
+After about a minute or so of running, the following message will appear, indicating that it has been successfully generated:
+
+![](jsb/generate-file-complete.png)
+
+After the above steps, all the files under **cocos/bindings/auto** will be automatically generated, which is very convenient.
+
+The following is an example of how the TS layer prints logs by calling the Native layer log method through JSB, and how to implement the auto-binding tool to generate the corresponding auto-binding files based on the C++ code written.
+
+## Writing the C++ layer implementation
+
+C++ is the bridge between TypeScript and Native layers. To implement JSB calls, the first step is to prepare the header files and implementation for the C++ layer, here we create a `test` folder in `cocos` directory to store the relevant files:
+
+![](jsb/store-file.png)
+
+`ABCJSBBridge.h`, declares an `abcLog` function for the Typescript layer to call the logging, and because the logging method will certainly be used in many places in the ts layer, a singleton pattern is used here, providing `getInstance()` to get the current instance of the class.
```

Comment:
```suggestion
`ABCJSBBridge.h`, declares an `abcLog` function for the TypeScript layer to call the logging, and because the logging method will certainly be used in many places in the ts layer, a singleton pattern is used here, providing `getInstance()` to get the current instance of the class.
```